From d2c14205434af3f74b82cf000fa7e73de0e57a29 Mon Sep 17 00:00:00 2001
From: Yuan Jianpeng <yuanjianpeng@360.cn>
Date: Mon, 16 Mar 2020 21:30:44 +0800
Subject: [PATCH] add out-of-tree build support

---
 hostapd/Makefile | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/hostapd/Makefile b/hostapd/Makefile
index 2a6bd7a..75ee290 100644
--- a/hostapd/Makefile
+++ b/hostapd/Makefile
@@ -27,7 +27,7 @@ CFLAGS += -I$(abspath ../src/utils)
 
 export BINDIR ?= /usr/local/bin/
 
--include .config
+include $(O).config
 
 ifndef CONFIG_NO_GITVER
 # Add VERSION_STR postfix for builds from a git repository
@@ -1271,7 +1271,7 @@ endif
 
 ALL=hostapd hostapd_cli
 
-all: verify_config $(ALL)
+all: $(if $(O),,verify_config) $(addprefix $(O),$(ALL))
 
 Q=@
 E=echo
@@ -1289,7 +1289,8 @@ ifdef CONFIG_CODE_COVERAGE
 	@$(E) "  CC " $<
 	$(Q)cd $(dir $@); $(CC) -c -o $(notdir $@) $(CFLAGS) $(notdir $<)
 else
-%.o: %.c
+$(O)%.o: %.c
+	$(Q)[ -d "$(dir $@)" ] || mkdir -p "$(dir $@)"
 	$(Q)$(CC) -c -o $@ $(CFLAGS) $<
 	@$(E) "  CC " $<
 endif
@@ -1316,15 +1317,15 @@ install: $(addprefix $(DESTDIR)$(BINDIR)/,$(ALL))
 
 BCHECK=../src/drivers/build.hostapd
 
-hostapd: $(BCHECK) $(OBJS)
-	$(Q)$(CC) $(LDFLAGS) -o hostapd $(OBJS) $(LIBS)
+$(O)hostapd: $(BCHECK) $(addprefix $(O),$(OBJS))
+	$(Q)$(CC) $(LDFLAGS) -o $@ $(addprefix $(O),$(OBJS)) $(LIBS)
 	@$(E) "  LD " $@
 
 ifdef CONFIG_WPA_TRACE
 OBJS_c += ../src/utils/trace.o
 endif
-hostapd_cli: $(OBJS_c)
-	$(Q)$(CC) $(LDFLAGS) -o hostapd_cli $(OBJS_c) $(LIBS_c)
+$(O)hostapd_cli: $(addprefix $(O),$(OBJS_c))
+	$(Q)$(CC) $(LDFLAGS) -o $@ $(addprefix $(O),$(OBJS_c)) $(LIBS_c)
 	@$(E) "  LD " $@
 
 NOBJS = nt_password_hash.o ../src/crypto/ms_funcs.o $(SHA1OBJS)
-- 
2.20.1

